person.rb

```diff
@@ -0,0 +1,65 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    original_name = @nameable.correct_name
+    original_name.length > 10 ? original_name[0..9] : original_name
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** These classes should be in their own files i.e. 4 files. This very file should only be for the Person class. I strongly suggested you look into this.